subsys/bluetooth/host/sdp.c

```diff
@@ -861,6 +874,47 @@ static uint8_t select_attrs(struct bt_sdp_attribute *attr, uint8_t att_idx,
 			     (space < seq_size + sad->cont_state_size))) {
 				/* Packet exhausted */
 				sad->state->pkt_full = true;
+
+				/* If packet is NULL, put part of the attribute */
+				if (sad->state->rsp_buf_empty) {
+					//if (!sad->state->pkt_full && sad->rsp_buf) {
+					/* construct the whole attribute data to sdp_max_att_pool */
+					struct net_buf *att_buf = net_buf_alloc(&sdp_max_att_pool, K_NO_WAIT);
+					if (att_buf != NULL) {
+						struct bt_sdp_data_elem_seq *att_seq;
+						if (!sad->seq &&
+							(sad->state->current_svc != sad->rec->index)) {
+								att_seq = net_buf_add(att_buf, sizeof(*att_seq));
```

Comment:
There are way too many levels of indentation here. Please look for ways to refactor this. Options include splitting stuff into header functions, inverting if-branch conditions (potentially combined with goto's and labels, etc).

Reply:
Thanks. I will try to simplify it later.